cluster-operator/src/main/java/io/strimzi/operator/cluster/model/EntityTopicOperator.java

```diff
@@ -249,8 +254,30 @@ public static EntityTopicOperator fromCrd(Kafka kafkaAssembly) {
             createVolumeMount(EntityOperator.TLS_SIDECAR_CA_CERTS_VOLUME_NAME, EntityOperator.TLS_SIDECAR_CA_CERTS_VOLUME_MOUNT));
     }
 
-    public RoleBindingOperator.RoleBinding generateRoleBinding(String namespace) {
-        return new RoleBindingOperator.RoleBinding(roleBindingName(cluster), EntityOperator.EO_CLUSTER_ROLE_NAME,
-                namespace, EntityOperator.entityOperatorServiceAccountName(cluster), createOwnerReference());
+    public KubernetesRoleBinding generateRoleBinding(String namespace) {
+        KubernetesSubject ks = new KubernetesSubjectBuilder()
+                .withKind("ServiceAccount")
+                .withName(EntityOperator.entityOperatorServiceAccountName(cluster))
+                .withNamespace(namespace)
+                .build();
+
+        KubernetesRoleRef roleRef = new KubernetesRoleRefBuilder()
+                .withName(EntityOperator.EO_CLUSTER_ROLE_NAME)
+                .withApiGroup("rbac.authorization.k8s.io")
+                .withKind("ClusterRole")
+                .build();
+
+        KubernetesRoleBinding rb = new KubernetesRoleBindingBuilder()
+                .withNewMetadata()
+                .withName(roleBindingName(cluster))
+                .withNamespace(namespace)
+                .withOwnerReferences(createOwnerReference())
+                .withLabels(labels.toMap())
```

Comment:
Do we need to check labels for null? `labels == null ? null : labels.toMap()`

Reply:
Do we do it for another resources? AFAIK the labels are never null because they contain at least the default `strimzi.io/...` labels. So it might not be needed.